torch/testing/_internal/logging_utils.py

```diff
@@ -130,6 +130,9 @@ def tearDownClass(cls):
         torch._logging._internal.log_state.clear()
         torch._logging._init_logs()
 
+    def hasRecord(self, records, m):
+        return any(m in r.getMessage() for r in records)
+
```

Comment:
What is this?

Reply:
There are two guards on jvp:
* One guarding JVP level
* Another one for forward ad level

`hasRecord` is used in the `test_jvp_guard_fail` to check which of the guards fail. On my machine it is usually the latter that fails, but I wrote this code to match for both in case guarding order is not deterministic.